### PR TITLE
implemented shape

### PR DESCRIPTION
**Pull request by haynes1**

### CODE REVIEW

File: src/s2/shape.rs [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+		if (v0cmp != 0) {
+			return v0cmp;
+		} else {
+			Cmp(e.V1, other.V1);
+		}
+	}
+}
+
+// sortEdges sorts the slice of Edges in place.
+pub fn sortEdges(e: Vec<Edge>) -> Vec<Edge> {
```
Comment (crgant12) [inline src/s2/shape.rs]: This probably needs to be done like this example https://rust-lang-nursery.github.io/rust-cookbook/algorithms/sorting.html

File: src/s2/shape.rs [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+	e.sort()
+}
+
+// edges implements the Sort interface for slices of Edge.
+type Edges = Vec<Edge>
+
+impl Edges {
+
+	pub fn Len(&self, ) -> i64 { len(self) }
```
Comment (crgant12) [inline src/s2/shape.rs]: This is for go's way of sorting things. You probably don't need this.

File: src/s2/shape.rs [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+
+	// IsEmpty reports whether the Shape contains no points. (Note that the full
+	// polygon is represented as a chain with zero edges.)
+	fn IsEmpty() -> bool;
+
+	// IsFull reports whether the Shape contains all points on the sphere.
+	fn IsFull() -> bool;
+
+	// We do not support implementations of this interface outside this package.
+	privateInterface()
```
Comment (crgant12) [inline src/s2/shape.rs]: I would just take this out since it doesn't translate directory from go.

File: src/s2/shape.rs [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+// Similarly, an Shape representing 5 points would have 5 chains consisting
+// of one edge each.
+//
+// Shape has methods that allow edges to be accessed either using the global
+// numbering (edge ID) or within a particular chain. The global numbering is
+// sufficient for most purposes, but the chain representation is useful for
+// certain algorithms such as intersection (see BooleanOperation).
+pub trait Shape {
+	// NumEdges returns the number of edges in this shape.
+	fn NumEdges() -> i64;
```
Comment (crgant12) [inline src/s2/shape.rs]: Rename all of these the rust way.

**Review by crgant12 — CHANGES_REQUESTED** [review]

File: src/r2/point.rs [hunk 5]
```diff
@@ -17,16 +17,63 @@ limitations under the License.
...
+#[derive(Clone, Copy, Debug)]
 pub struct Point {
     /// x coordinate of the point
     pub x: f64,
     /// y coordinate of the point
     pub y: f64,
 }
 
+
+impl Eq for Point {}
```
Comment (crgant12) [inline src/r2/point.rs:31]: Delete this since you're not implementing it.

File: src/r2/point.rs [hunk 6]
```diff
@@ -17,16 +17,63 @@ limitations under the License.
...
+        Some(self.cmp(other))
+    }
+}
+
+impl PartialEq for Point {
+    fn eq(&self, other: &Point) -> bool {
+        self.cmp(other) == Ordering::Equal
+    }
+}
+
```
Comment (crgant12) [inline src/r2/point.rs:64]: Delete all of the blank lines

**Review by crgant12 — CHANGES_REQUESTED** [review]

File: src/r2/point.rs (hunk 5 above)
Comment (haynes1) [reply src/r2/point.rs:31]: This needs to stay. I agree its weird that you have to impl something that has no methods
https://doc.rust-lang.org/std/cmp/trait.Eq.html

**Review by crgant12 — APPROVED** [review]

**Pull request merged**